packages/block-editor/src/components/block-mover/index.native.js

```diff
@@ -10,58 +11,123 @@ import { ToolbarButton } from '@wordpress/components';
 import { __, sprintf } from '@wordpress/i18n';
 import { withSelect, withDispatch } from '@wordpress/data';
 import { withInstanceId, compose } from '@wordpress/compose';
-import { arrowUp, arrowDown } from '@wordpress/icons';
+import { arrowUp, arrowDown, arrowLeft, arrowRight } from '@wordpress/icons';
+
+const horizontalMover = {
+	backwardButtonIcon: arrowLeft,
+	forwardButtonIcon: arrowRight,
+	backwardButtonHint: __( 'Double tap to move the block to the left' ),
+	forwardButtonHint: __( 'Double tap to move the block to the right' ),
+	firstBlockTitle: __( 'Move block left' ),
+	lastBlockTitle: __( 'Move block right' ),
+	/* translators: accessibility text. %1: current block position (number). %2: next block position (number) */
+	backwardButtonTitle: __(
+		'Move block left from position %1$s to position %2$s'
+	),
+	/* translators: accessibility text. %1: current block position (number). %2: next block position (number) */
+	forwardButtonTitle: __(
+		'Move block right from position %1$s to position %2$s'
+	),
+};
+
+const verticalMover = {
+	backwardButtonIcon: arrowUp,
+	forwardButtonIcon: arrowDown,
+	backwardButtonHint: __( 'Double tap to move the block up' ),
+	forwardButtonHint: __( 'Double tap to move the block down' ),
+	firstBlockTitle: __( 'Move block up' ),
+	lastBlockTitle: __( 'Move block down' ),
+	/* translators: accessibility text. %1: current block position (number). %2: next block position (number) */
+	backwardButtonTitle: __( 'Move block up from row %1$s to row %2$s' ),
+	/* translators: accessibility text. %1: current block position (number). %2: next block position (number) */
+	forwardButtonTitle: __( 'Move block down from row %1$s to row %2$s' ),
+};
 
 const BlockMover = ( {
 	isFirst,
 	isLast,
+	isRTL,
 	isLocked,
 	onMoveDown,
 	onMoveUp,
 	firstIndex,
 	rootClientId,
+	horizontalDirection,
 } ) => {
+	const {
+		backwardButtonIcon,
+		forwardButtonIcon,
+		backwardButtonHint,
+		forwardButtonHint,
+		firstBlockTitle,
+		lastBlockTitle,
+	} = horizontalDirection ? horizontalMover : verticalMover;
+
 	if ( isLocked || ( isFirst && isLast && ! rootClientId ) ) {
 		return null;
 	}
 
+	const switchButtonPropIfRTL = (
+		isBackwardButton,
+		forwardButtonProp,
+		backwardButtonProp
+	) => {
+		if ( isRTL && horizontalDirection ) {
+			// for RTL and horizontal direction switch prop between forward and backward button
+			if ( isBackwardButton ) {
+				return forwardButtonProp; // set forwardButtonProp for backward button
+			}
+			return backwardButtonProp; // set backwardButtonProp for forward button
+		}
+
+		return isBackwardButton ? backwardButtonProp : forwardButtonProp;
+	};
```

Comment:
I have introduce this function because it was common for title and icon when we have RTL mode and horizontal direction (we need to switch prop between buttons)

Does its name make sense ?

Reply:
I think it does make sense. I checked the VoiceOver in RTL mode as well, and it is working fine thanks to this.